src/env0-environments-provider.ts

```diff
@@ -106,13 +107,22 @@ export class Env0EnvironmentsProvider
   }
 
   async refresh(): Promise<void> {
-    if (this.isRefreshing) {
+    if (this.isRefreshing || !stateManager.isLoggedIn) {
       return;
     }
     this.isRefreshing = true;
     try {
       const newEnvironments = await getEnvironmentsForBranch();
-      if (this.shouldUpdate(newEnvironments) && this.isRefreshing) {
+      // we need to avoid taking any action in case user logged out while we were waiting for the response
+      if (!stateManager.isLoggedIn) {
+        return;
+      }
+      if (newEnvironments.length === 0) {
+        stateManager.setNoEnvironment(true);
+      } else {
+        stateManager.setNoEnvironment(false);
+      }
```

Comment:
```suggestion
        stateManager.setNoEnvironment(newEnvironments.length === 0);
      }```

Reply:
```suggestion
       stateManager.setNoEnvironment(_.isEmpty(newEnvironments));
```
is even better

Reply:
a644f33e6ec87b1b6d421e178c4858d48d44fb2f